### PR TITLE
Add support for `.log(level, msg)` calls in `flake8-logging-format`

### PR DESCRIPTION
**Pull request by dhruvmanila**
fixes: #2562

### CODE REVIEW

File: crates/ruff/resources/test/fixtures/flake8_logging_format/G001.py [hunk 1]
```diff
@@ -2,4 +2,5 @@
 import logging as foo
 
 logging.info("Hello {}".format("World!"))
+logging.log(logging.INFO, "Hello {}".format("World!"))
```
Comment (charliermarsh) [inline crates/ruff/resources/test/fixtures/flake8_logging_format/G001.py:5]: Can we add a test case for, e.g., `logging.log(msg="foo", level=logging.INFO)`?

Comment (charliermarsh) [reply crates/ruff/resources/test/fixtures/flake8_logging_format/G001.py:5]: Actually I got it, looks like the code works correctly :)

File: crates/ruff/src/rules/flake8_logging_format/rules.rs [hunk 2]
```diff
@@ -146,13 +163,17 @@ pub fn logging_call(checker: &mut Checker, func: &Expr, args: &[Expr], keywords:
...
+            if let Some(format_arg) = call_args.argument("msg", msg_pos) {
                 check_msg(checker, format_arg);
             }
 
             // G010
             if checker.settings.rules.enabled(Rule::LoggingWarn)
-                && matches!(logging_level, LoggingLevel::Warn)
+                && matches!(
+                    logging_call_type,
+                    LoggingCallType::LevelCall(LoggingLevel::Warn)
```
Comment (charliermarsh) [inline crates/ruff/src/rules/flake8_logging_format/rules.rs:175]: I wonder if we should extend this to warn on `logging.log(logging.WARN, "...")`

Comment (charliermarsh) [reply crates/ruff/src/rules/flake8_logging_format/rules.rs:175]: (Can be in a separate PR if needed, not even sure whether that's correct to flag.)

Comment (dhruvmanila) [reply crates/ruff/src/rules/flake8_logging_format/rules.rs:175]: Regarding the level variable, I'm not sure as the docs doesn't mention anything about it although the `WARN` level isn't mentioned in the [levels](https://docs.python.org/3/library/logging.html#logging-levels) table.

If we do decide to change it, it'll need to be updated wherever the `level` argument is accepted.

**Pull request merged**